HQSmokeTests/testCases/conftest.py

```diff
@@ -0,0 +1,154 @@
+            "download.default_directory": str(UserInputsData.download_path),
+            "download.prompt_for_download": False,
+            "download.directory_upgrade": True,
+            "safebrowsing.enabled": True})
+    else:
+        chrome_options.add_argument('--safebrowsing-disable-download-protection')
+        chrome_options.add_argument('--safebrowsing-disable-extension-blacklist')
+        # #
+        # chrome_options.add_argument("--window-size=1920,1080")
+        # chrome_options.add_argument('--start-maximized')
+        # chrome_options.add_argument('--no-sandbox')
+        # chrome_options.add_argument("--disable-extensions")
+        # chrome_options.add_argument('--headless')
+
+        chrome_options.add_experimental_option("prefs", {
+            "download.default_directory": str(UserInputsData.download_path),
+            "download.prompt_for_download": False,
+            "safebrowsing.enabled": True})
+    web_driver = ChromeDriverManager().install()
+
+    driver = webdriver.Chrome(executable_path=web_driver, options=chrome_options)
+    request.cls.driver = driver
+    login = LoginPage(driver, settings["url"])
+    login.login(settings["login_username"], settings["login_password"])
+    
+    yield driver
+
+    driver.close()
+    driver.quit()
+    
+
+@pytest.hookimpl(hookwrapper=True)
+# @pytest.mark.hookwrapper
+def pytest_runtest_makereport(item):
+    print("entering report formation")
+    pytest_html = item.config.pluginmanager.getplugin("html")
+    outcome = yield 
+    report = outcome.get_result()
+    extra = getattr(report, 'extra', [])
+   
+    if report.when == "call" or report.when == "teardown": 
+        print(report.when)
+        xfail = hasattr(report, 'wasxfail')
+        if (report.skipped and xfail) or (report.failed and not xfail):
+            print("report skipped or failed")
+            file_name = report.nodeid.replace("::", "_") + ".png" 
+            #file_name = None
+            screen_img = _capture_screenshot(item.funcargs['init_driver'])
+            if file_name:
+                html = '<div><img src="data:image/png;base64,%s" alt="screenshot" style="width:600px;height:300px;" ' \
+                       'onclick="window.open(this.src)" align="right"/></div>' % screen_img
+                extra.append(pytest_html.extras.html(html))
+        report.extra = extra
+        print("extra added to report")
+   
+def _capture_screenshot(driver):
+    return driver.get_screenshot_as_base64()
+
+@pytest.fixture
+def email_pytest_report(req):
+    "pytest fixture for device flag"
+    return req.config.getoption("--email_pytest_report")
+
+# Command line options:
+def pytest_addoption(parser):
+    parser.addoption("--email_pytest_report",
+                 dest="email_pytest_report",
+                 help="Email pytest report: Y or N",
+                 default="Y")
+
+def pytest_terminal_summary(terminalreporter, exitstatus):
```

Comment:
this seems a bit complex, would it not be easier to just run this as a separate build step after the tests are complete?